Shopify/hydrogen
# Support Remix v2 in route generator

Related: https://github.com/Shopify/hydrogen/pull/747

### WHY are these changes introduced?

Remix has future flags that change the way routes work in a project: `v2_routeConvention`, `v2_meta` and `v2_errorBoundary`.
So far, our generator logic only supports v1.

### WHAT is this pull request doing?

With this PR, we check if any of the supported flags is enabled in the project and change the way we generate new routes.
If `v2_routeConvention` is enabled, the new files are generated in the corresponding path for v2. For example, `routes/_index.tsx` instead of `routes/index.tsx`.
If `v2_meta` or `v2_errorBoundary` are enabled, the corresponding exports in route files are adapted to v2 format (see the test file for more info).

Since v2 future flags are now supported in full in Hydrogen, I've also enabled these flags in the `hello-world` template so that new projects start using Remix v2 features.



### HOW to test your changes?

From the hello-world template, run `npx shopify hydrogen generate routes` and check that they all work (v1). Then, remove all the routes, enable the mentioned flags in `remix.config.js#future`, and run the command again. The files should have different filepaths, and some of them should contain v2 version of the meta export (policies and page).

### Expected review

Would like to know:
- Other approaches that come to your mind?
- Any issue with this approach or concerns for the future?
- Otherwise, code-review to merge this asap.

#### Checklist
